curriculum/challenges/english/25-front-end-development/lab-one-time-password-generator/67c562286b29447da020d407.md

```diff
@@ -0,0 +1,341 @@
+challengeType: 25
+dashedName: build-a-one-time-password-generator
+demoType: onClick
+---
+
+# --description--
+
+In this lab, you will generate a 6-digit OTP (One-Time Password) and display it to the user. The OTP will expire after 5 seconds, and a new OTP will be generated when the user clicks the `"Generate New OTP"` button.
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. You should use the `useEffect` hook to manage the countdown timer.
+2. Your `OTPGenerator` component should return a `div` element with the class name `container`.
+3. The `.container` `div` should include the following elements:
+
+    - An `h1` element with the ID `otp-title` and text `"OTP Generator"`.
+    - An `h2` element with the ID `otp-display` that either displays the message `"Click 'Generate   OTP' to get a code"` or shows the generated OTP if one is available.
+    - A `p` element with the ID `otp-timer` that:
+        - Starts off empty.
+        - Displays `"Expires in: X seconds"` after the button is clicked, where `X` represents the remaining time before the OTP expires.
+        - Shows the message `"OTP expired. Click the button to generate a new OTP."` once the countdown reaches `0`.
+    - A `button` element with the ID `generate-otp-button` labeled `"Generate OTP"`. When clicked, it should generate a new OTP and start a 5-second countdown.
+    - The `"Generate OTP"` button must be disabled while the countdown is active.
+
+4. You should ensure the countdown timer stops automatically once it reaches `0` seconds to prevent unnecessary updates.
+5. The generated OTP should be 6 digits long.
+
+# --hints--
+
+You should use the `useEffect` hook.
+
+```js
+// to be added
+```
+
+You should not remove the existing `const { useState, useEffect } = React;` assignment from the code.
+
+```js
+assert.match(code, /{\s*(useState\s*,\s*useEffect|useEffect\s*,\s*useState|useState)\s*}\s*=\s*React;/);
+```
+
+You should render a `div` with the class `.container`.
+
+```js
+const containerDiv = document.querySelector('.container');
+assert.exists(containerDiv);
+```
+
+The `.container` should contain an `h1` element with the ID `otp-title` the text `"OTP Generator"`.
+
+```js
+const h1 = document.querySelector(".container h1#otp-title");
+assert.exists(h1);
+assert.equal(h1.textContent, "OTP Generator");
+```
+
+The `.container` should contain an `h2` element with the ID `otp-display` that displays the OTP when generated. 
+
+```js
+const h2 = document.querySelector(".container h2#otp-display");
+assert.exists(h2);
+```
+
+Initially, the `h2` inside `.container` should display the message `"Click 'Generate OTP' to get a code"`.
+
+```js
+assert.strictEqual(document.querySelector('.container h2#otp-display').textContent.trim(), "Click 'Generate OTP' to get a code");
+```
+
+The `.container` should contain a `p` element with the ID `otp-timer` that is initially empty.
```

Comment:
```suggestion
Initially, the `.container` should not contain any `p` elements.
```
The solution starts without them and only shows one after the button is clicked.